pkg/kube/resourcerequirements/resource_requirements.go

```diff
@@ -0,0 +1,49 @@
+package resourcerequirements
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+const (
+	resourceMemory = "memory"
+	resourceCpu    = "cpu"
+)
+
+// New returns a new corev1.ResourceRequirements with the specified arguments, and an error
+// which indicates if there was a problem parsing the input
+func New(limitsCpu, limitsMemory, requestsCpu, requestsMemory string) (corev1.ResourceRequirements, error) {
```

Comment:
is this worth passing a struct here so that it's easier to understand what value you pass, and maybe address the use-case of passing no value and taking the default?

![](https://media.giphy.com/media/NEJCPYzr6xmAU/giphy.gif)

Reply:
or maybe a builder (again) that comes with default values and you can call .setLimitsCpu(xxx) and so on

Reply:
I'm on a little bit on the fence about this one. When we're not doing much (just 4 values). I like the simplicity of a single factory-style function. It might make sense to introduce a builder when we need to start . configuring non-defaults here.

Reply:
as discuss, un-export New for now, and we'll see later if we need a builder